inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -97,80 +71,87 @@ def gen_config(self, simulator: Simulator):
         )
 
     @singledispatchmethod
-    def run_simulation(self, simulator: Simulator):
+    def gen_pipeline(self, simulator: Simulator):
+        pass
+
+    @singledispatchmethod
+    def gen_aux_files(self, simulator: Simulator, input_dir: str):
```

Comment:
Why raising this error here ?

I do think we need to throw this exception if the `simulator` argument in the `run` method is not GROMACS (i.e. someone tries to run this scenario passing in a another simulator)

Reply:
The exception throw can be removed from here, but the function has to exist, such that the registration below (see line 84) works.

For reference, this was introduced when we added `singledispatchmethods` to handle the methods `gen_aux_files`, in order to avoid the solution presented [here](https://github.com/inductiva/inductiva/pull/154). See [this](https://github.com/inductiva/inductiva/pull/203). It's a weird (and ugly!) solution, so it would be great to come up with an alternative...